src/stan/services/pathfinder/single.hpp

```diff
@@ -820,7 +820,16 @@ inline auto pathfinder_lbfgs_single(
         logger.info(lbfgs_ss);
         lbfgs_ss.str("");
       }
-      throw e;
+      if (ReturnLpSamples) {
+        // we want to terminate multi-path pathfinder during these unrecoverable
+        // exceptions
+        throw;
```

Comment:
Why? One pathfinder can fail while the others succeed?

Reply:
> If one path does something unrecoverable like indexes into a vector out of bounds, I think the right thing to do is stop and say "something is clearly wrong with your model".

Are only unrecoverable errors able to be thrown here? Like if the call to log_prob throws it could be from something recoverable like one of the ode solvers failing for a given set of parameters

Reply:
`pathfinder_impl` catches `std::domain_error`s, which is what the ODE solvers throw (and is generally what the math library considers 'recoverable')